### ISSUE TITLE
Preserve list numbering in Markdown source

### ISSUE THREAD

**Issue by Boldewyn** [gnab/remark#512]
Often I find myself in a situation, where I continue a previous `<ol>` on the next slide:

```
1. Foo
2. Bar

---

3. Quux
4. Quax
```

Currently I have to resort to plain HTML with `<ol start="3">` to achieve this layout.

It would be great, if remark had an option to enable preserved list numbers from the Markdown source.

marked has a pull request, that implemented that feature: https://github.com/markedjs/marked/pull/127/files
Unfortunately it was closed during the current try to revive the project. But maybe it's possible to somewhat easily use the code from there and inject it in remark's version of marked?

**Comment by abelards** [comment]
Well, it's hard to say because this is a markdown feature: "don't care about numbers".

Personally when working on my slides I will just move lines around or duplicate lines, so I end up having 1 2 2 3 4 and I'm glad markdown will fix it to 1 2 3 4 5.

Also, I hope that on slides you will likely never reach more than 10 bullet points so it's not so hard to ask people to re-number their bullets, but still I like to be lazy :)

So I think this will be useful for 5% of our users, but 95% of our users will be sad.
What do you all think?

**Comment by Boldewyn** [comment]
I agree, that the Markdown default of not caring about the concrete numbers is a blessing in most MD documents I write. Unfortunately, in this one concrete case of lists being split on several slides, it stands in the way. Therefore a default-off option would be great that would allow such a use-case, while itself not getting in the way of all the other users, who trust on Markdown’s default behaviour.

**Comment by abelards** [comment]
I think the `<ol start="x">` is great, and perhaps I'll have a look deeper so we can make that X something where you can just say "hey, look at that counter here" so even if you add bullet #3 to slide 1, it continues numbering #4 on your slide 2 (even if the number you typed is 3).

We can even use named counters, CSS's `counter-increment` and `counter-reset` and `OL LI::before { content: counter(my-counter) }` so that you could do "OLs inside OLs" without messing things up (chapter counters, title counters, subtitles counters...).

I'm not sure how we would add options in a clean markup-ish way though, which is why I'd prefer  if we stayed in full CSS, like

```
# slide 1
.carry-over-counter[
  1. one
  2. two
  2. three
]
---
# slide 2
.carry-over-counter[
  3. four
  4. five
]
```
